### ISSUE TITLE
Its not possible to create a DM or invite a user with a full MXID on a new account

### ISSUE THREAD

**Issue by pixlwave** [vector-im/element-x-ios#1500]
### Steps to reproduce

1. Use OIDC to create a new account.
2. Click the prominent Start Chat button.
3. Enter a full mxid to tell your colleague "wooo, it worked, I'm signed into Element"

#### What did you expect?

To be able to start a chat.

#### What happened instead?

![Simulator Screen Recording - iPhone 14 - 2023-08-16 at 15 14 02](https://github.com/vector-im/element-x-ios/assets/6060466/deb9578c-fa3d-448b-acc9-2a6984bfff5f)

Homeserver Ts & Cs not accepted.
```
2023-08-16T14:00:42.162374Z DEBUG root:send{homeserver="https://my.homeserver/" server_versions=[V1_0, V1_1, V1_2, V1_3, V1_4, V1_5, V1_6] config=RequestConfig { timeout: 30s } request_id="REQ-27" request_size="197 B" path="/_matrix/client/v3/createRoom" status=403 response_size="389 B"}: matrix_sdk::http_client: crates/matrix-sdk/src/http_client/mod.rs:208: Error while sending request: Api(Server(ClientApi(Error { status_code: 403, authenticate: None, body: Standard { kind: _Custom { errcode: "M_CONSENT_NOT_GIVEN", extra: Extra({"consent_uri": String("https://my.homeserver/_matrix/consent?u=myuserid&h=6757a23ece6b4347206dabe6eadd5cf652ac75db4ca10af5061ca1b83151ebd3")}) }, message: "To continue using this homeserver you must review and agree to our terms and conditions at https://my.homeserver/_matrix/consent?u= myuserid&h=6757a23ece6b4347206dabe6eadd5cf652ac75db4ca10af5061ca1b83151ebd3." } })))
```

### Your phone model

Simulator

### Operating system version

16.4

### Application version

1.2.4 `develop`

### Homeserver

element.io

### Will you send logs?

No

**Comment by hughns** [comment]
I believe this is a server configuration problem rather than something that an OIDC native client should handle. I will clarify and confirm.

**Comment by hughns** [comment]
The homeserver shouldn't be using the consent feature of the Matrix Client-Server API and instead it should be handled by the OIDC Provider.

This is now enforced in Synapse by https://github.com/matrix-org/synapse/pull/16127 which is included in >=1.91.0.
